ocs_ci/deployment/deployment.py

```diff
@@ -0,0 +1,292 @@
+"""
+This module provides base class for different deployment
+platforms like AWS, VMWare, Baremetal etc.
+"""
+
+import os
+import logging
+import time
+import json
+
+import pytest
+import yaml
+
+from ocs_ci.framework import config
+from ocs_ci.utility import templating, system
+from ocs_ci.ocs.utils import create_oc_resource, apply_oc_resource
+from ocs_ci.utility.utils import (
+    run_cmd, ceph_health_check,
+)
+from ocs_ci.ocs.exceptions import CommandFailed
+from ocs_ci.ocs import constants, ocp, defaults
+from ocs_ci.ocs.resources.ocs import OCS
+from tests import helpers
+
+
+logger = logging.getLogger(name=__file__)
+
+
+class Deployment(object):
+    """
+    Base for all deployment platforms
```

Comment:
I think OCP and OCS dont have much in common, probably they dont have to be mixed here, we can take this up as new PR.

Reply:
@vasukulkarni I'm not sure what you mean here, can you elaborate? We are separating OCP and OCS deployment steps in this base class.

Reply:
the deployment class seems to be the base class for ocp and ocs, which shoudn't be the case since they are two different things and putting them in same deployment base class doesn't really makes sense.

Reply:
`AWSBase` is the only class that inherits from `Deployment`

Reply:
I have added an issue here: https://github.com/red-hat-storage/ocs-ci/issues/483 , I will also add this in trello with flow diagram.